### PR TITLE
Sound gui graph theme mapping automation

### PR DESCRIPTION
**Pull request by Stormgun**
This is the auto mapper of themes to sceneGraph structure, the only todo left is to unit test the configurator itself for a fully completed job

### CODE REVIEW

File: src/js/components/basic-draggable/drag-drop-container.jsx [hunk 1]
```diff
@@ -15,11 +15,15 @@ var ItemGroupForThemeUnion = React.createClass({
             if (parent[property].type === 'city') {
                 classNames += property;
             }
-
-            if (parentList.length < 2) {
-                classNames += " col-lg-4"; //If root nodes add into bootstrap column system.
+            if(parent[property].type === 'root'){
+                classNames += "dol-lg-" + 12/parentList.length;
```
Comment (aaronpcz) [inline src/js/components/basic-draggable/drag-drop-container.jsx]: Looks like a typo dol-lg? I assume this should be col-lg

File: src/js/utils/scene-graph/graphTheme-generator.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+var SceneStore = require('../../stores/scene-store')
+var _ = require('lodash');
+
+module.exports = {
+    generateGraphThemes: function (sceneGraph) {
```
Comment (aaronpcz) [inline src/js/utils/scene-graph/graphTheme-generator.js]: Would be nice with a few simple mocha unit tests

File: src/js/utils/scene-graph/category-config-generator.js [hunk 3]
```diff
@@ -36,18 +36,19 @@ module.exports = {
...
-            var columnHeaders = getUniqArrByName(columnHeaders);
-            if (prevConfig) {
-                var rowGen = transferPastGenerationAliases(prevConfig.rowHeaders, rowHeaders);
-                rowGen= getUniqArrByName(rowGen);
-                var columnGen = transferPastGenerationAliases(prevConfig.columnHeaders, columnHeaders);
-                columnGen= getUniqArrByName(columnGen);
-                config = {rowHeaders: rowGen, columnHeaders: columnGen, themes: themes};
-            } else {
-                config = {rowHeaders: rowHeaders, columnHeaders: columnHeaders, themes: themes};
+            if(rowHeaders !== [] && columnHeaders !==[]) {
```
Comment (aaronpcz) [inline src/js/utils/scene-graph/category-config-generator.js]: Would be nice with a few simple mocha unit tests

File: src/js/utils/HubClient.js [hunk 4]
```diff
@@ -8,7 +8,8 @@ var assetStore = require('./asset-store');
 var connectionCache = require('./connection-cache');
 var NodeListGeneration = require('./scene-graph/node-list-generation');
 var GraphTypes = require('../constants/graph-constants').GraphTypes;
-var CategoryConfig = require('../utils/scene-graph/category-config-generator')
+var CategoryConfig = require('../utils/scene-graph/category-config-generator');
+var GraphThemeGenerator = require('../utils/scene-graph/graphTheme-generator');
```
Comment (aaronpcz) [inline src/js/utils/HubClient.js]: A tiny point but the new file name would probably suit graph-theme-generator ? just following largely precedent (yes we do have a few exceptions but they are largely old files)

File: src/js/components/basic-draggable/drag-drop-container.jsx (hunk 1 above)
Comment (Stormgun) [reply src/js/components/basic-draggable/drag-drop-container.jsx]: Yep saw it

**Pull request merged**